lib/reverse_sentence.rb

```diff
@@ -1,6 +1,20 @@
 # A method to reverse the words in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n)
+# Space complexity: O(n)
+def split_my_sentence(sentence)
+  sentence.scan(/\S+|\s/)
+end
+
 def reverse_sentence(my_sentence)
```

Comment:
⚠️ This solution works, but it's O(n^2) time complexity and O(n) space complexity.  It can be done in-place, i.e. O(1) space complexity, and O(n) time complexity.  

Consider:  Go through and swap each character in the string with it's opposite number on the other side of the string, thus reversing the string.  Then go through and reverse each individual word the same way.  The trick is figuring out where words start and end.